articles/migrations/guides/legacy-lock-api-deprecation.md

```diff
@@ -0,0 +1,195 @@
+
+The purpose of this guide is to help you to select the best migration path for your application(s) if you are impacted by the deprecation notice. 
+
+## Am I affected?
+
+If your applications match any of the following cases, you are affected:
+
+* Use of versions of Lock previous to v11 and versions of Auth0.js previous to v9 in embedded login scenarios
+* Use of /usernamepassword/login endpoint directly from applications 
+* Use of /user/ssodata endpoint directly from applications
+
+If you do not use the above libraries and do not specifically call the above endpoints, you are not affected. No libraries which are not specifically named are affected by this vulnerability, or in turn, by the migration.
+
+### If you already use Universal Login / Hosted Login Page
+
+Applications which log users in via Universal Login through an Auth0 hosted page are not _required_ to update the version of Lock or Auth0.js that they use _inside_ that login page (if you have customized your login page in the [Dashboard](${manage_url}/#/login_page). However, the use of the newest library versions is strongly recommended, even in the Universal Login Page. For those who have not customized their login page, the Lock v11 widget is already in use and no further action is required.
+
+### If you use embedded login
+
+Embedded login with Lock v11 and Auth0.js v9 now rely entirely on [cross-origin authentication](/cross-origin-authentication) for any form of username/password authentication (such as Auth0 DB, Custom DB, or LDAP connections). Cross-origin authentication is the situation in which credentials are collected on one domain, but validated on another.
+
+This cross-origin authentication protocol relies on cookies, which will be considered third-party cookies if the domain of the application and Auth0 tenant do not match. Unfortunately, some browsers block third-party cookies, and even if supported, many users may have manually disabled third-party cookies in their browsers. 
+
+Because of these [cross-origin authentication issues](/cross-origin-authentication#limitations-of-cross-origin-authentication), there are only two general implementations that can be recommended. 
+
+1. [Migrate to Universal Login](#1-migrate-to-universal-login). Universal Login will work with or without [custom domains](/custom-domains), and will work from most application types as well. It requires the least application code to implement and is the most secure option. 
+
+2. [Continue to use embedded login and migrate to newer library versions](#2-continue-to-use-embedded-login-and-migrate). A crucial part of this implementation is employing a custom domain to prevent cross-origin issues. The following caveats apply:
+    * Only one custom domain can be applied per Auth0 tenant, so all applications on the tenant will use the same custom domain. 
```

Comment:
Instead of "so all applications on the tenant will use the same custom domain. ", I'd say "so, if you want to have Single Sign On between multiple applications, all need to be hosted in the same top level domain"

Reply:
@jeffreylees just edited the previous one, instead of replying to it, sorry, so if you acted on it, please check it again ;)